src/actions/posts.js

```diff
@@ -920,6 +932,47 @@ export function getNeededAtMentionedUsernames(state, posts) {
     return usernamesToLoad;
 }
 
+export function getNeededCustomEmojis(state, posts) {
+    let customEmojisByName; // Populate this lazily since it's relatively expensive
+    const nonExistentEmoji = state.entities.emojis.nonExistentEmoji;
+
+    const customEmojisToLoad = new Set();
+
+    Object.values(posts).forEach((post) => {
+        if (!post.message.includes(':')) {
+            return;
+        }
+
+        if (!customEmojisByName) {
+            customEmojisByName = selectCustomEmojisByName(state);
+        }
+
+        const pattern = /\B:([A-Za-z0-9_-]+):/gi;
```

Comment:
This needs another `\B` at the end to be consistent with the markdown parser so that it doesn't catch things like `:strawberry:sundae`

`/\B:([A-Za-z0-9_-]+):\B/gi`